lib/src/builder/generation/impl_generation.dart

```diff
@@ -652,9 +660,37 @@ class ImplGenerator {
         ..writeln())
         .toString();
   }
+
+  String _generateConsumablePropsOrStateClass(AccessorType type, NodeWithMeta node) {
+    if (!(node is PropsOrStateNode)) {
+      return '';
+    }
+
+    if ((node as PropsOrStateNode).hasCompanionClass) {// ignore: avoid_as
+      return '';
+    }
+
+    final className = _classNameFromNode(node);
+    final typeParameters = (node.node as ClassDeclaration).typeParameters;
+    final typeParamsOnClass = typeParameters?.toSource() ?? '';
+    final typeParamsOnSuper = removeBoundsFromTypeParameters(typeParameters);
+    final consumableClassName = _publicPropsOrStateClassNameFromConsumerClassName(className);
+
+    final classKeywords = '${type == AccessorType.abstractProps || type == AccessorType.abstractState ? 'abstract ' : ''}class';
+    final isProps = type == AccessorType.props || type == AccessorType.abstractProps;
+    return (StringBuffer()
+      ..writeln('$classKeywords $consumableClassName$typeParamsOnClass extends $className$typeParamsOnSuper with ${className}AccessorsMixin$typeParamsOnSuper {')
+      ..writeln(_generateStaticMetaDecl(consumableClassName, isProps))
+      ..writeln('}'))
+        .toString();
+  }
+
+  String _generateStaticMetaDecl(String consumableClassName, bool isProps) {
+    return '  static const ${isProps ? 'Props' : 'State'}Meta meta = ${_metaConstantName(consumableClassName)};';
+  }
 }
 
-enum AccessorType {props, state}
+enum AccessorType {props, state, abstractProps, abstractState}
```

Comment:
There are still a handful of places where there are checks like:
```dart
final isProps = type == AccessorType.props;
```
Even if they're not problematic right now, they might be brittle against future changes.

What about changing this to something like:
```dart
class AccessorType {
  final bool isProps;
  final bool isAbstract;
  const AccessorType(bool isProps, bool isAbstract);

  static const AccessorType props = const AccessorType(true, false);
  static const AccessorType state = const AccessorType(false, false);
  static const AccessorType abstractProps = const AccessorType(true, true);
  static const AccessorType abstractState = const AccessorType(false, true);
}
```
and updating those checks:
```diff
-type == AccessorType.props
+type.isProps
```